cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +454,243 @@ RetryLoop:
+type labelFilter struct {
+	GadgetName string
+	NodeName string
+	Namespace string
+	PodName string
+	ContainerName string
+	OutputMode string
+}
+
+// labelsFromFilter creates a string containing labels value from the given
+// labelFilter.
+func labelsFromFilter(filter labelFilter) string {
+	labels := ""
+	separator := ""
+
+	v := reflect.ValueOf(filter)
+	typeOfS := v.Type()
+
+	// Loop on all fields of labelFilter.
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i).Interface()
+
+		// If this field has no value, just skip it.
+		if field == "" {
+			continue
+		}
+
+		// If labels is not empty, we will need to separate the other label value
+		// with a comma, so change separator value.
+		if labels != "" {
+			separator = ","
+		}
+
+		// Concatenate the label to existing one.
+		labels = fmt.Sprintf("%s%s%s=%v", labels, separator, typeOfS.Field(i).Name, v.Field(i).Interface())
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	return labels
+}
+
+// getTraceListFromParameters returns traces associated with the given config.
+func getTraceListFromParameters(config *TraceConfig) ([]gadgetv1alpha1.Trace, error) {
+	namespace := ""
+	if !config.CommonFlags.AllNamespaces {
+		namespace = GetNamespace()
+	}
+
+	filter := labelFilter{
+		GadgetName: config.GadgetName,
+		NodeName: config.CommonFlags.Node,
+		Namespace: namespace,
+		PodName: config.CommonFlags.Podname,
+		ContainerName: config.CommonFlags.Containername,
+		OutputMode: config.TraceOutputMode,
+	}
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: labelsFromFilter(filter),
+		FieldSelector: fields.Everything().String(),
+	}
+
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return []gadgetv1alpha1.Trace{}, err
+	}
+
+	return traces.Items, nil
+}
+
+// PrintAllTraces prints all traces corresponding to the given config.CommonFlags.
+func PrintAllTraces(config *TraceConfig) error {
+	traces, err := getTraceListFromParameters(config)
+	if err != nil {
+		return err
+	}
+
+	type printingInformation struct {
+		namespace string
+		nodeName string
+		podname string
+		containerName string
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+
+	fmt.Fprintln(w, "NAMESPACE\tNODE(S)\tPOD\tCONTAINER\tTRACEID")
+
+	printingMap := map[string]printingInformation{}
+
+	for _, trace := range traces {
+		labels := trace.ObjectMeta.Labels
+
+		id, present := labels[TRACE_ID]
+		if !present {
+			return errors.New("Trace has no ID!")
+		}
+
+		info, present := printingMap[id]
+		if present {
+			node := labels["NodeName"]
+			if node == "" {
+				continue
+			}
+
+			// If an entry with this traceID already exists, we just update the node
+			// name by concatenating it to the string.
+			var builder strings.Builder
+
+			builder.WriteString(info.nodeName)
+			builder.WriteString(",")
+			builder.WriteString(node)
+
+			info.nodeName = builder.String()
```

Comment:
```suggestion
			info.nodeName = info.nodeName + "," + node
```

Does it work fine too?